### ISSUE TITLE
[terra-image] Require alt text prop.

### ISSUE THREAD

**Issue by mjhenkes** [cerner/terra-core#3396]
# Feature Request

## Description
The alt text should be required, not optional. Documentation should be added to indicate that decorative images should have an alt text of ' ', but consumers should be required to think about what text to add.

A notice should be added indicating that aria labels should not be applied.

Add an a11y guide for implementation of the onLoad/OnError props. @scottwilmarth do you have more details about what a11y concerns exist over using onLoad/onError?

## Additional Context / Screenshots

Audit: #3079
